pkg/cli/sso.go

```diff
@@ -0,0 +1,216 @@
+package cli
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/convox/convox/pkg/structs"
+	"github.com/convox/convox/sdk"
+	"github.com/convox/convox/sso"
+	"github.com/convox/stdcli"
+)
+
+func init() {
+	registerWithoutProvider("sso configure", "configure sso config file", SsoConfigure, stdcli.CommandOptions{
+		Flags: []stdcli.Flag{
+			stdcli.StringFlag("provider", "p", "sso provider"),
+			stdcli.StringFlag("client_id", "c", "client id"),
+			stdcli.StringFlag("client_secret", "s", "client secret"),
+			stdcli.StringFlag("issuer", "i", "issuer"),
+		},
+		Validate: stdcli.ArgsMax(0),
+	})
+
+	registerWithoutProvider("sso login", "authenticate with a console using sso", SsoLogin, stdcli.CommandOptions{
+		Validate: stdcli.ArgsMax(0),
+	})
+}
+
+func SsoConfigure(rack sdk.Interface, c *stdcli.Context) error {
+	var err error
+	provider := coalesce(c.String("provider"), os.Getenv("SSO_PROVIDER"))
+	if provider == "" {
+		c.Writef("SSO Provider: ")
+
+		provider, err = c.ReadSecret()
+		if err != nil {
+			return err
+		}
+
+		c.Writef("\n")
+	}
+
+	clientID := coalesce(c.String("client_id"), os.Getenv("SSO_CLIENT_ID"))
+	if clientID == "" {
+		c.Writef("SSO Client ID: ")
+
+		clientID, err = c.ReadSecret()
+		if err != nil {
+			return err
+		}
+
+		c.Writef("\n")
+	}
+
+	clientSecret := coalesce(c.String("client_secret"), os.Getenv("SSO_CLIENT_SECRET"))
+	if clientSecret == "" {
+		c.Writef("SSO Client Secret: ")
+
+		clientSecret, err = c.ReadSecret()
+		if err != nil {
+			return err
+		}
+
+		c.Writef("\n")
+	}
+
+	issuer := coalesce(c.String("issuer"), os.Getenv("SSO_ISSUER"))
+	if issuer == "" {
+		c.Writef("SSO ISSUER: ")
+
+		issuer, err = c.ReadSecret()
+		if err != nil {
+			return err
+		}
+
+		c.Writef("\n")
+	}
```

Comment:
the option to set the environment variables is not documented, can you add it plz?